src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -7,29 +7,27 @@
  */
 
 import {
-  Directive,
-  Input,
-  ContentChildren,
-  QueryList,
   AfterContentInit,
-  OnDestroy,
-  Optional,
-  NgZone,
+  Directive,
   ElementRef,
   Inject,
+  NgZone,
+  OnDestroy,
+  Optional,
   Self,
 } from '@angular/core';
 import {Directionality} from '@angular/cdk/bidi';
-import {FocusKeyManager, FocusOrigin} from '@angular/cdk/a11y';
-import {LEFT_ARROW, RIGHT_ARROW, UP_ARROW, DOWN_ARROW, ESCAPE, TAB} from '@angular/cdk/keycodes';
-import {takeUntil, mergeAll, mapTo, startWith, mergeMap, switchMap} from 'rxjs/operators';
-import {Subject, merge} from 'rxjs';
+import {DOWN_ARROW, ESCAPE, LEFT_ARROW, RIGHT_ARROW, TAB, UP_ARROW} from '@angular/cdk/keycodes';
+import {takeUntil} from 'rxjs/operators';
 import {CdkMenuGroup} from './menu-group';
-import {CDK_MENU, Menu} from './menu-interface';
-import {CdkMenuItem} from './menu-item';
-import {MenuStack, MenuStackItem, FocusNext, MENU_STACK} from './menu-stack';
+import {CDK_MENU} from './menu-interface';
+import {FocusNext, MENU_STACK, MenuStack} from './menu-stack';
 import {PointerFocusTracker} from './pointer-focus-tracker';
-import {MenuAim, MENU_AIM} from './menu-aim';
+import {MENU_AIM, MenuAim} from './menu-aim';
+import {CdkMenuBase} from './menu-base';
+
+// TODO(mmalerba): Keyboard controls should match
+//   https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-1/menubar-1.html#
```

Comment:
I did some testing with this and found that it mostly matches what it should be. I did find 2 issues:

Issue 1 (Menu Bar Example):
- click "File"
- press "e"
- focuses "Edit" but does not close "File" menu
- should close "File" menu

Issue 2 (Menu Bar example):
- click "File"
- press down
- press left
- closes menu and focuses "File"
- should focus "Edit"

I also noticed that in the a11y demo, menu items were triggered by hover rather than click. Personally I hate hover triggered menus (used to drive me crazy on Amazon) but should we have a configuration option for people to enable that?

Reply:
Fixed issue 2. For issue 1 (the typeahead), drive does close the menu when doing typeahead, and that feels probably like what we should do. However I think I will need to add an API to our key manager to support that. I'll do it a in a followup PR